scene/main/window.cpp

```diff
@@ -647,6 +648,12 @@ void Window::_update_viewport_size() {
 	_set_size(final_size, final_size_override, attach_to_screen_rect, stretch_transform, allocate);
 
 	if (window_id != DisplayServer::INVALID_WINDOW_ID) {
+#ifdef OSX_ENABLED
```

Comment:
Not sure this is the right way and place to downscale it. Font oversampling probably should be changed too.